aiida/restapi/translator/base.py

```diff
@@ -399,7 +323,29 @@ def set_query(
 
         ## Add projections
         if projections is None:
-            self.set_default_projections()
+            if attributes is None and extras is None:
+                self.set_default_projections()
+            else:
+                default_projections = self.get_default_projections()
+                if attributes in [True, 'true', 'True']:
+                    if attributes_filter is None:
+                        default_projections.append('attributes')
+                    ## Check if attributes_filter is a string or a list
+                    elif isinstance(attributes_filter, six.string_types):
+                        default_projections.append('attributes.' + attributes_filter)
+                    elif isinstance(attributes_filter, list):
+                        for attr in attributes_filter:
+                            default_projections.append('attributes.' + attr)
```

Comment:
As discussed elsewhere, I wouldn't return `attributes.xxx` but a dictionary `attributes` just with the filtered keys. See also comment about why we still have both attributes_filter and alist (if I understand the logic, I would just keep attributes_filter, add some other to replace `nalist`, and if attributes_filter is not specified I would assume it means no filter => all attributes).

Reply:
Changed to return `attributes` or `extras` dictionary with only the filtered keys.
The filters `alist` and `nalist` are instead used in the `/nodes/<id>/contents/attributes/` endpoint.